ConEmuWinForms/ConEmuControl.cs

```diff
@@ -130,14 +103,14 @@ public bool IsStatusbarVisible
 
 		/// <summary>
 		///     <para>Starts a new console process in the console emulator control, and shows the console emulator view. When a session is not running, the control is blank.</para>
-		///     <para>If another <see cref="RunningSession">session</see> is running, it will be closed, and the new session will replace it. <see cref="AutoStartInfo" /> will call <see cref="Start" /> and create a session automatically if configured.</para>
+		///     <para>If another <see cref="RunningSession">session</see> is running, it will be closed, and the new session will replace it.</para>
 		/// </summary>
 		/// <remarks>
 		///     <para>The control state transitions to <see cref="States.ConsoleEmulatorWithConsoleProcess" />, then to <see cref="States.ConsoleEmulatorEmpty" /> (unless configured to close on exit), then to <see cref="States.Recycled" />.</para>
 		/// </remarks>
 		/// <returns>Returns the newly-started session, as in <see cref="RunningSession" />.</returns>
 		[NotNull]
-		public ConEmuSession Start([NotNull] ConEmuStartInfo startinfo)
+		public ConEmuSession Start([NotNull] ConEmuStartInfo startinfo, [NotNull] JoinableTaskFactory joinableTaskFactory)

```

Comment:
@AArnott Do we have a best practice for a library that doesn't define JTF itself but wants to work within an application that uses it?

Reply:
Yes...'ish.
You can take a `JoinableTaskContext` as a parameter to some initialization method, or a `JTF` as you've done here is fine. We haven't done this a lot (but recently we've done it in a couple places). A property instead of a constructor parameter is another way we've gone.

One option to consider is allowing null in your library, and if it is null, assume your caller is on the UI thread and create a JTC instance then and there to serve the library so that the rest of the code can assume there is one, even if the app didn't give it a 'singleton' for the app. That sounds nice for the library, but if every library did that, an app might suffer from forgetting to pass the JTC to some of the libraries and as a result, hit deadlocks until they find them all. So I'm not sure how well it scales.